pinot-core/src/main/java/com/linkedin/pinot/core/common/DataFetcher.java

```diff
@@ -32,19 +32,35 @@
 public class DataFetcher {
   private static final BlockId BLOCK_ZERO = new BlockId(0);
 
-  private final IndexSegment _indexSegment;
+  // private final IndexSegment _indexSegment;
 
-  private final Map<String, DataSource> _columnToDataSourceMap = new HashMap<>();
-  private final Map<String, Dictionary> _columnToDictionaryMap = new HashMap<>();
-  private final Map<String, BlockValSet> _columnToBlockValSetMap = new HashMap<>();
+  private final Map<String, BaseOperator> _columnToDataSourceMap;
```

Comment:
We want to avoid operators accessing api's on datasource. Accessing datasource directly makes it hard to add support for optional dictionary.